src/renderer/components/Login.js

```diff
@@ -144,28 +168,22 @@ class Login extends React.Component {
                 type='text'
                 value={addr}
                 leftIcon='envelope'
-                onChange={this.handleChange}
+                onChange={this.handleCredentialsChange}
               />
             </FormGroup>
             <FormGroup label={tx('login.mailPw')} placeholder='Password' labelFor='mailPw' labelInfo={`(${tx('login.required')})`}>
-              <InputGroup
-                id='mailPw'
-                leftIcon='lock'
-                type='mailPw'
-                value={mailPw}
-                onChange={this.handleChange}
-              />
+              {this.renderPasswordInput('showPasswordMail', 'mailPw')}
             </FormGroup>
-            <Button onClick={this.handleShowAdvanced}>{(this.state.showAdvanced ? '-' : '+') + ' ' + tx('login.advanced') }</Button>
-            <Collapse isOpen={this.state.showAdvanced}>
+            <Button onClick={this.handleUISwitchStateProperty.bind(this, 'showAdvanced')}>{(showAdvanced ? '-' : '+') + ' ' + tx('login.advanced') }</Button>
```

Comment:
I wonder if this also should be internationalized? but it's probably good enough for now